src/modules/hip/handlehip.cpp

```diff
@@ -303,6 +303,9 @@ Handle::Handle() : impl(new HandleImpl())
 #endif
     this->SetAllocator(nullptr, nullptr, nullptr);
     impl->PreInitializeBuffer();
+    impl->numThreads = std::min(impl->numThreads, std::thread::hardware_concurrency());
```

Comment:
I see this change has been made in handlehip and handlehost. But the hip handle is created with "rppCreateWithStreamAndBatchSize". Our RPP build with HIP backend can run HOST code too. We need to ensure same host thread behavior during runtime on both backend builds.

Reply:
In handlehip.cpp, there are 2 type constructor overloads for Handle creation
`Handle::Handle(rppAcceleratorQueue_t stream, size_t batchSize) : impl(new HandleImpl())`
https://github.com/sampath1117/rpp/blob/sr/num_threads_update/src/modules/hip/handlehip.cpp#L251

This constructor is called when rppCreateStreamWithBatchSize is called from outer api and RPP is built with HIP backend

Another constructor is
Handle::Handle(size_t batchSize, Rpp32u numThreads) : impl(new HandleImpl())
https://github.com/sampath1117/rpp/blob/sr/num_threads_update/src/modules/hip/handlehip.cpp#L282

This constructor is called when rppCreateWithBatchSize is called from outer api and RPP is built with HIP backend
Here i have added the numThreads changes in this constructor. This constructor is being called when calling from HOST test suite and RPP is built with HIP backend

From test suite i have ensured the changes done in handlehip.cpp will be reflected for HOST code too when RPP is built with HIP backend

So i don't think we need to change other HIP constructor where stream is given as input

Reply:
Okay sounds good! thanks